### DISCUSSION TITLE
Session Key Decryption Failed

### DISCUSSION THREAD

**Discussion by Wajid-01** [openpgpjs/openpgpjs#1697] [Q&A]
Hi,

I have a key-pair, which I'm currently implementing for file encryption. We pass the public key to SFMC, who encrypt the csv and send it to us (in `<filename>.csv.gpg format.` Our Lambda (using typescript) then decrypts the csv using the private key.

**The following behaviour occurs:**

1. Oddly, when I encrypt the csv locally (using the gpg command), and have the Lambda decrypt using private key - it works.

2. When SFMC encrypt and the lambda decrypts using private key, it does not. With the error: 

"Error decrypting message: Session key decryption failed. Error decrypting message: Session key decryption failed.\n"

**In the logs, I noticed the following variance in encryption key Ids of the encrypted files:**

When I locally encrypt:
- [ Ke { bytes: 'VµrVÕ×ÑY' } ].

When SFMC encrypts:
- [ Ke { bytes: 'Î\x80ã\x17U\r\x19\x0E' } ]

HOWEVER, both of the above Ids, exist in the private key Ids:

- [ Ke { bytes: 'Î\x80ã\x17U\r\x19\x0E' }, Ke { bytes: 'VµrVÕ×ÑY' } ]

Please advise.

**Comment by larabr** [comment]
Could you check if this works: https://github.com/openpgpjs/openpgpjs/discussions/1352#discussioncomment-931652

**Comment by Wajid-01** [comment]
Sadly, it doesn't. It gives me another error: "message is not
authenticated". I read you could previously disable MDC, as that's the
cause of the above error, but it was removed in later releases (not sure
it'd be a good idea to remove, even if it was possible).

I'm also confused, as openpgp is so widely used. Unsure why SFMC would
default to using the master key as opposed to the subkey.

Could you advise further please?

**Reply by larabr** [reply]
To fix the "Message not authenticated error", you need to also pass `config.allowUnauthenticatedMessages = true` when decrypting. This is in addition to the config flag in https://github.com/openpgpjs/openpgpjs/discussions/1352#discussioncomment-931652 .

> Unsure why SFMC would default to using the master key as opposed to the subkey.

I am not familiar with their implementation, but it wouldn't be the first nor only app/library that does that, unfortunately (hence the config flag 😄 ). Based on the message authentication error, they seem to be relying on a legacy implementation of OpenPGP.
